test/Microsoft.TestCommon/PlatformInfo.cs

```diff
@@ -23,6 +24,12 @@ public static Platform Platform
 
         private static Platform GetPlatform()
         {
+            if (Type.GetType(_netCore20TypeName, throwOnError: false) != null)
+            {
+                // Treat .NET Core 2.0 as a .NET 4.5 superset though internal types are different.
+                return Platform.Net45;
```

Comment:
No need for a runtime check whether test is running on .NET Core. Only the test assembly built for .NET Core is run on .NET Core.